endpoints/users.js

```diff
@@ -5,24 +5,40 @@ const config = require('../config');
 
 create = (request, response) => {
     if (!request.body.username) {
-        response.status(400).json({"err": "Missing username field."});
-    }
-    if (!request.body.password) {
-        response.status(400).json({"err": "Missing password field."});
-    }
+        response.status(400).json({"err": "Missing username field."}).end();
+    } else if (!request.body.password) {
+        response.status(400).json({"err": "Missing password field."}).end();
+    } else if (request.body.username.length < 5){
+        response.status(400).json({"err": "Username must be 5 or more characters."}).end();
+    } else {
 
-    bcrypt.hash(request.body.password, 10, function(err, hash) {
-        if (err) {
-            response.status(500).end();
+        let lengthPassed = (request.body.password.length >= 7); 
+        let containsCap = /[A-Z]/.test(request.body.password);
+        let containsNum = /\d/.test(request.body.password);
+        let containsSymbol = /\W/.test(request.body.password);
+        if (!lengthPassed || !containsCap || !containsNum || !containsSymbol) {
+            let error = {"err" : ""};
+            error.err += (lengthPassed ? "" : "Password must be 7 or more characters. ");
```

Comment:
I dislike this. I wouldn't want an API that I use to return a long string with all of the errors in it. It appears that this code will return something like `{"err": "Password must be 7 or more characters. Password must contain a capital. Password must contain a number. "}` What I would expect from this endpoint is instead `{"err": ["Password must be...", "err2", err3", "etc..."]}`. A JSON array of all of the errors.